File: tools/python/maton_agent_toolkit/configuration.py
"""Configuration types for Maton Agent Toolkit."""

from typing_extensions import TypedDict


class Context(TypedDict, total=False):
    """Context for MCP connection."""
    connection: str | None
    mode: str | None


class Configuration(TypedDict, total=False):
    """Configuration for Maton Agent Toolkit."""
    context: Context | None

File: tools/typescript/src/shared/configuration.ts
// Context are settings that are applied to all requests made by the integration.
export type Context = {
  // Connection is a Maton connection ID. If set, the integration will route
  // requests through this connection (sent as the Maton-Connection header).
  connection?: string;

  // If set to 'modelcontextprotocol', the Maton MCP calls will use a special
  // User-Agent header.
  mode?: 'modelcontextprotocol' | 'toolkit';
};

// Configuration provides various settings and options for the integration
// to tune and manage how it behaves.
export type Configuration = {
  context?: Context;
};
